apps/src/applab/AppLabView.jsx

```diff
@@ -34,8 +35,101 @@ var AppLabView = React.createClass({
     onMount: React.PropTypes.func.isRequired
   },
 
+  getInitialState() {
+    // only used so that we can rerender when resized
+    return {
+      width: undefined,
+      height: undefined
+    };
+  },
+
+  /**
+   * Called when the window resizes. Look to see if width/height changed, then
+   * call adjustHeights as our maxHeight may need adjusting.
+   */
+  onResize: function () {
+    var windowWidth = $(window).width();
+    var windowHeight = $(window).height();
+
+    // We fire window resize events when the grippy is dragged so that non-React
+    // controlled components are able to rerender the editor. If width/height
+    // didn't change, we don't need to do anything else here
+    if (windowWidth === this.state.width && windowHeight === this.state.height) {
+      return;
+    }
+
+    this.setState({
+      width: $(window).width(),
+      height: $(window).height()
```

Comment:
Small nit here: The naming makes it unclear whether you intend to capture the window dimensions, or the dimensions of `AppLabView` (which are different, since it renders into a space below the header).  Maybe inconsequential for how you're using it right now, but I wouldn't want someone to come along later, see `AppLabView.state.height` and think it represented the height of the component.